plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/dataimport/MappingTargetCreator.java

```diff
@@ -0,0 +1,108 @@
+ *******************************************************************************/
+package org.eclipse.fordiac.ide.model.dataimport;
+
+import java.util.List;
+
+/*******************************************************************************
+ * Copyright (c) 2024 Primetals Technologies Austria GmbH
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Alois ZOitl - initial implementation
+ *******************************************************************************/
+import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.eclipse.fordiac.ide.model.helpers.BlockInstanceFactory;
+import org.eclipse.fordiac.ide.model.libraryElement.ConfigurableFB;
+import org.eclipse.fordiac.ide.model.libraryElement.FB;
+import org.eclipse.fordiac.ide.model.libraryElement.FBNetworkElement;
+import org.eclipse.fordiac.ide.model.libraryElement.LibraryElementFactory;
+import org.eclipse.fordiac.ide.model.libraryElement.Resource;
+import org.eclipse.fordiac.ide.model.libraryElement.TypedSubApp;
+import org.eclipse.fordiac.ide.model.libraryElement.UntypedSubApp;
+import org.eclipse.fordiac.ide.model.libraryElement.VarDeclaration;
+import org.eclipse.fordiac.ide.model.typelibrary.FBTypeEntry;
+import org.eclipse.jdt.annotation.NonNull;
+
+public final class MappingTargetCreator {
+
+	private MappingTargetCreator() {
+		throw new UnsupportedOperationException("Utility class shall not be instantated!"); //$NON-NLS-1$
+	}
+
+	public static FBNetworkElement createMappingTarget(final Resource res, final FBNetworkElement srcElement,
+			final String targetFBName) {
+		final FBNetworkElement created = createFBNetworkElement(srcElement);
+
+		if (created != null) {
+			if (srcElement.getTypeEntry() != null) {
+				created.setTypeEntry(srcElement.getTypeEntry());
+			}
+			created.setInterface(srcElement.getInterface().copy()); // use the src interface to get all parameters
+			created.setPosition(EcoreUtil.copy(srcElement.getPosition()));
+			if (srcElement instanceof final ConfigurableFB srcConfFB) {
+				setupConfigureableFB(srcConfFB, (ConfigurableFB) created);
+			}
+			created.setName(extractTargetFBName(targetFBName));
+			res.getFBNetwork().getNetworkElements().add(created);
+		}
+		return created;
+	}
+
+	public static void transferFBParams(@NonNull final FBNetworkElement srcElement,
+			@NonNull final FBNetworkElement targetElement) {
+		final List<VarDeclaration> destInputs = targetElement.getInterface().getInputVars();
+		final List<VarDeclaration> srcInputs = srcElement.getInterface().getInputVars();
+
+		for (int i = 0; i < destInputs.size(); i++) {
+			final VarDeclaration srcVar = srcInputs.get(i);
+			final VarDeclaration dstVar = destInputs.get(i);
+
+			if ((null != srcVar.getValue()) && (!srcVar.getValue().getValue().isEmpty())) {
```

Comment:
Please be consistent with you null checks.
in line 53 you write srcElement.getTypeEntry() != null and here youwrite null != ....

Reply:
Can't you use the change value command here ?

Reply:
No I can not use any commands in the model plugin.

Reply:
fixed